src/AutoRest.CSharp/Mgmt/Decorator/InheritanceChoser.cs

```diff
@@ -147,8 +147,16 @@ private static bool IsEqual(System.Type parentType, ObjectTypeProperty[] propert
                         !IsAssignable(parentProperty.PropertyType, childPropertyType) &&
                         !(parentProperty.PropertyType.IsGenericParameter && IsAssignable(parentProperty.PropertyType.BaseType!, childPropertyType)))
                     {
-                        //TODO(ADO item 5712): deal with protected setter
-                        return false;
+                        if (parentProperty.PropertyType.ContainsGenericParameters && parentProperty.PropertyType.BaseType != null && IsAssignable(parentProperty.PropertyType.BaseType, childPropertyType))
```

Comment:
should be removable now with nisha's changes

Reply:
done